### PR TITLE
feat(inference): add mean-field variational inference scaffold

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add first mean-field variational inference scaffold for Rubix inference
- implement diagonal Gaussian posterior initialization, sampling, and KL helper
- add variational optimization loop with reparameterized Monte Carlo objective
- add unit tests for KL properties, sampling, and VI optimization behavior

## Stack position
PR 7 / 8 in the gradient-production plan

### CODE REVIEW

File: rubix/inference/variational.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        key, step_key = jax.random.split(key)
+        value, grads = jax.value_and_grad(objective_only)(variational_params, step_key)
+        updates, opt_state = optimizer.update(grads, opt_state, variational_params)
+        variational_params = optax.apply_updates(variational_params, updates)
+
+        _, (reconstruction_value, kl_value) = objective_fn(variational_params, step_key)
+        objective_history.append(float(value))
+        reconstruction_history.append(float(reconstruction_value))
+        kl_history.append(float(kl_value))
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/variational.py]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/204#discussion_r3101111939)

File: rubix/inference/variational.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        best_posterior_mean_constrained = apply_transforms(
+            params=best_mean,
+            transforms=transforms,
+            direction="forward",
+        )
+
+    return VariationalResult(
+        posterior_mean_params=_tree_to_dict(posterior_mean_constrained),
+        posterior_log_std_params=_tree_to_dict(final_log_std),
+        best_posterior_mean_params=_tree_to_dict(best_posterior_mean_constrained),
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/variational.py]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/204#discussion_r3101111962)

File: rubix/inference/variational.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+        _, (reconstruction_value, kl_value) = objective_fn(variational_params, step_key)
+        objective_history.append(float(value))
+        reconstruction_history.append(float(reconstruction_value))
+        kl_history.append(float(kl_value))
+
+        if value < best_objective:
+            best_objective = value
+            best_mean = variational_params["mean"]
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/variational.py]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/204#discussion_r3101111987)
